src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -209,7 +207,11 @@ StartMetadataSyncToNode(const char *nodeNameString, int32 nodePort)
 	}
 
 	UseCoordinatedTransaction();
-	MarkNodeHasMetadata(nodeNameString, nodePort, true);
+
+	workerNode = SetWorkerColumn(workerNode, Anum_pg_dist_node_metadatasynced,
```

Comment:
It feels unexpected to first set `Anum_pg_dist_node_metadatasynced` and then `Anum_pg_dist_node_hasmetadata`. 

Why don't we do the other order?